File: sieve2.cs
using System.Collections;
class Sieve {
  static public void Main() {
    System.Console.WriteLine("C# (MS C Sharp)");
    const int Expected = 15485863;
    int count = 1; int i;
    const int N = 16000000; const int FINDPRIME = 1000000;
    int N_sqrt = (int) System.Math.Sqrt(N);
    BitArray b = new BitArray(N+1);
    for (i = 3; i <= N_sqrt; i += 2) {
      if ( b.Get(i) ) continue;
      count++;
      for (int k = i*i; k <= N; k += 2 * i) b.Set(k,true);
    }
    while ( i <= N ) {
      if ( ! b.Get(i) ) count++;
      if (count >= FINDPRIME) break;
      i += 2;
    }
    System.Console.WriteLine("Total Primes = " + count);
    System.Console.WriteLine("Max Prime = " + i);
    if (i != Expected) System.Console.WriteLine("ERROR: Largest Prime was incorrect. Should be: " + Expected);
  }
}

File: sieve2.rs
fn main() {
    println!("Rust");
    const EXPECTED: usize = 15485863;
    const FLOT: f64 = 16000000_f64;
    const N: usize = 16000000;
    const FINDPRIME: usize = 1000000;
    let n_sqrt: usize = FLOT.sqrt() as usize + 1;
    let mut count: usize = 1;
    let mut b = vec![0u8; 16000000];
    let mut i: usize = 3;
    let mut k: usize;

    while i <= n_sqrt {
        if b[i] == 0 {
            count += 1;
            k = i;
            while k <= N {
                b[k] = 1;
                k = k + (2 * i);
            }
        }
        i = i + 2;
    }

    while i <= N {
        if b[i] == 0 {
            count += 1;
            if count >= FINDPRIME {
                break;
            }
        }
        i = i + 2;
    }
    println!("Total Primes = {}", count);
    println!("Max Prime = {}", i);
    if i != EXPECTED {
        println!(
            "ERROR: Largest Prime was incorrect. Should be: {}",
            EXPECTED
        )
    }
}
